sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -31,11 +31,22 @@ object InterpretedPredicate {
   def create(expression: Expression, inputSchema: Seq[Attribute]): InterpretedPredicate =
     create(BindReferences.bindReference(expression, inputSchema))
 
-  def create(expression: Expression): InterpretedPredicate = new InterpretedPredicate(expression)
+  def create(expression: Expression): InterpretedPredicate = {
+    val predicate = new InterpretedPredicate(expression)
+    predicate.initialize(0)
```

Comment:
we not need InterpretedPredicate.initialize and modify _object InterpretedPredicate.create_
```
  def create(expression: Expression): InterpretedPredicate = {
    expression.foreach {
      case n: Nondeterministic => n.initialize(0)
      case _ =>
    }
    new InterpretedPredicate(expression)
  }
```